### ISSUE TITLE
rollup@0.25.2 can't build react-router

### ISSUE THREAD

**Issue by TrySound** [rollup/rollup#534]
Since 0.25.2 I get this error
```
Cannot overwrite the same content twice: 'invariant'
Error: Cannot overwrite the same content twice: 'invariant'
    at Object.patch (D:\_station\nvm\v0.12.7\node_modules\rollup\www\ROLLUP\rollup\node_modules\magic-string\dist\magic-string.es6.js:424:12)
    at Object.overwrite (D:\_station\nvm\v0.12.7\node_modules\rollup\www\ROLLUP\rollup\node_modules\magic-string\dist\magic-string.es6.js:393:8)
    at D:\_station\nvm\v0.12.7\node_modules\rollup\www\ROLLUP\rollup\src\Module.js:516:19
    at Array.forEach (native)
    at D:\_station\nvm\v0.12.7\node_modules\rollup\www\ROLLUP\rollup\src\Module.js:489:25
    at Array.forEach (native)
    at Module.render (D:\_station\nvm\v0.12.7\node_modules\rollup\www\ROLLUP\rollup\src\Module.js:429:19)
    at D:\_station\nvm\v0.12.7\node_modules\rollup\www\ROLLUP\rollup\src\Bundle.js:222:26
    at Array.forEach (native)
    at Bundle.render (D:\_station\nvm\v0.12.7\node_modules\rollup\www\ROLLUP\rollup\src\Bundle.js:221:23)
```

**Comment by TrySound** [comment]
Ref https://github.com/rollup/rollup/issues/502

**Comment by STAH** [comment]
https://github.com/rollup/rollup-plugin-replace/issues/4

**Comment by TrySound** [comment]
My config
```js
import nodeResolve from 'rollup-plugin-node-resolve';
import commonjs from 'rollup-plugin-commonjs';
import replace from 'rollup-plugin-replace';

export default {
  entry: 'src/main.js',
  external: [
    'react',
    'react-dom'
  ],
  plugins: [
    nodeResolve({ jsnext: true }),
    commonjs(),
    replace({
      'process.env.NODE_ENV': JSON.stringify('production')
    }),
  ],
  dest: 'public/main.js',
  format: 'iife',
  sourceMap: true
};
```

Code for reproducing
```js
import { Router, Route, IndexRoute, hashHistory } from 'react-router';

Router();
Route();
IndexRoute();
hashHistory();
```

**Comment by TrySound** [comment]
Yeah. It can be fixed in `replace`, but something in rollup@0.25.2 broke this.

**Comment by TrySound** [comment]
magic-string is affected
https://github.com/rollup/rollup/commit/366b414151542e4a8de1e6ba0d4ba0e560a0e78a
https://github.com/rollup/rollup/commit/1a0011485e912295df28bea534fd69abc823b701

/cc @Rich-Harris @eventualbuddha

**Comment by STAH** [comment]
Just add `&& node.consequent.type !== 'ConditionalExpression'`in https://github.com/rollup/rollup/commit/1a0011485e912295df28bea534fd69abc823b701#diff-5c98da346b849e07de8c1173579789b0R313 to skip optimization?

**Comment by Rich-Harris** [comment]
Minimal (or close to) reproduction:

```js
// main.js
import foo from './foo.js';
export default true ? false ? foo() : foo() : foo();

// foo.js
var foo = function () {};
export default Math.random() < 0.5 ? foo : foo;
```

Don't have time to look further into this now, but this is something to work off.

**Comment by heavyk** [comment]
this just happened to me. same thing but without a conditional. I'll work it down to a minimal test case and get back to you.

**Comment by eventualbuddha** [comment]
I verified that the example provided by @TrySound now works with #567 applied. If that looks good to everybody I'll merge and do a release.

**Comment by eventualbuddha** [comment]
Fixed in v0.25.6.
